From 898a51693ffaf5b796fc917fefddedeb8d651c35 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Tue, 14 Oct 2025 19:00:15 +0200
Subject: [PATCH] Call parent class on state event

This fixes the problem that maximize and fullscreen buttons didn't work
when client-side decoration is used.
---
 src/main-win.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/main-win.c b/src/main-win.c
index ce2bbdd..3447e54 100644
--- a/src/main-win.c
+++ b/src/main-win.c
@@ -573,6 +573,9 @@ void on_size_allocate( GtkWidget* widget, GtkAllocation    *allocation )
 
 gboolean on_win_state_event( GtkWidget* widget, GdkEventWindowState* state )
 {
+#if GTK_CHECK_VERSION(3, 0, 0)
+    GTK_WIDGET_CLASS(main_win_parent_class)->window_state_event( widget, state );
+#endif
     MainWin* mw = (MainWin*)widget;
     if( (state->new_window_state & GDK_WINDOW_STATE_FULLSCREEN) != 0 )
     {
